plugins/clipboard/plugin.js

```diff
@@ -2539,6 +2580,227 @@
 			return undefined;
 		}
 	};
+
+	/**
+	 * Fallback dataTransfer object which is used together with {@link CKEDITOR.plugins.clipboard.dataTransfer}
+	 * for browsers supporting Clipboard API, but not supporting custom MIME types (Edge 16+ - #962).
+	 *
+	 * @since 4.8
+	 * @class CKEDITOR.plugins.clipboard.fallbackDataTransfer
+	 * @constructor Creates a class instance.
+	 * @param {Object} [nativeDataTransfer] A native data transfer object.
+	 * @param {String} [customDataFallbackType=text/html] A MIME type used for storing custom data types.
+	 */
+	CKEDITOR.plugins.clipboard.fallbackDataTransfer = function( nativeDataTransfer, customDataFallbackType ) {
+
+		/**
+		 * A native dataTransfer object.
+		 *
+		 * @private
+		 * @property {Object} _nativeDataTransfer
+		 */
+		this._nativeDataTransfer = nativeDataTransfer;
+
+		/**
+		 * A MIME type used for storing custom MIME types. Defaults to 'text/html'.
+		 *
+		 * @private
+		 * @property {String} _customDataFallbackType
+		 */
+		this._customDataFallbackType =  customDataFallbackType || 'text/html';
+	};
+
+	/**
+	 * True if the environment supports custom MIME types in {@link CKEDITOR.plugins.clipboard.dataTransfer#getData}
+	 * and {@link CKEDITOR.plugins.clipboard.dataTransfer#setData} methods. Introduced to distinguish browsers which
+	 * supports only some whitelisted types (like 'text/html', 'application/xml'), but does not support custom
+	 * MIME types (like `cke/id`) like Edge 16+. When the value of this property equals 'null' it means it was not yet initialized.
+	 * This property should not be accessed directly, use {@link #isRequired} method instead.
+	 *
+	 * @private
+	 * @static
+	 * @property {Boolean}
+	 */
+	CKEDITOR.plugins.clipboard.fallbackDataTransfer._isCustomMimeTypeSupported = null;
+
+	CKEDITOR.plugins.clipboard.fallbackDataTransfer.prototype = {
+		/**
+		 * Whether {@link CKEDITOR.plugins.clipboard.fallbackDataTransfer} should be used when operating on dataTransfer.
+		 * If true is returned, it means custom MIME types are not supported in the current browser (see {@link #_isCustomMimeTypeSupported}).
+		 *
+		 * @returns {Boolean}
+		 */
+		isRequired: function() {
+			// If there is no `dataTransfer` we cannot detect if fallback is needed.
+			// Method returns `false` so regular flow will be applied.
+			if ( !this._nativeDataTransfer ) {
```

Comment:
I noticed there is one not handled edge case here, if `fallbackDataTransfer._isCustomMimeTypeSupported` is already initialized (let's say with `true`). But then we will get an event without native data transfer, `isRequired` will return `false` as a result of the first check:
```
if ( !this._nativeDataTransfer ) {
	return false;
}
```

So the above check should also be inside `if ( fallbackDataTransfer._isCustomMimeTypeSupported === null ) {` to prevent such situation.